### PR TITLE
Consume Bookstore  API

### PR DESCRIPTION
**Pull request by rosemutai**
On this PR:

- I made use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as the backend.
- I used [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make HTTP requests to the backend.
- I fetched  books from the Bookstore API after the component has rendered.
- I refactored the `add book` and `remove book` features to persist  changes made in the server.

### CODE REVIEW

File: src/components/AddBook.jsx [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
 
   const submitForm = (e) => {
     e.preventDefault();
 
-    const id = books.length + 1
-
-    dispatch(addBook({ item_id: `item${id}`, title, author }));
+    dispatch(addNewBook({ title, author, category: "" }));
+    dispatch(fetchBooks())
   }
```
Comment (ShadyShawkat) [inline src/components/AddBook.jsx]: - [x] Kindly make sure that the app works properly. Whenever a task is created or removed the whole list disappears from the DOM. kindly fix this.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @rosemutai, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/AddBook.jsx [hunk 2]
```diff
@@ -1,21 +1,18 @@
...
-  const books = useSelector((state) => state.books.books)
 
   const submitForm = (e) => {
     e.preventDefault();
 
-    const id = books.length + 1
-
-    dispatch(addBook({ item_id: `item${id}`, title, author }));
+    dispatch(addNewBook({ title, author, category: "" }));
   }
```
Comment (DammyShittu) [inline src/components/AddBook.jsx:16]: - **[OPTIONAL]** I think it would a good idea to clear the inputs after adding a book.

**Review by DammyShittu — APPROVED** [review]
Hi @rosemutai,

## APPROVED 🟢

## Highlights
✔️ API feature

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**